app/src/main/java/com/example/hodhod/hotels/viewmodels/HotelViewModel.java

```diff
@@ -0,0 +1,68 @@
+package com.example.hodhod.hotels.viewmodels;
+
+import android.util.Log;
+
+import com.example.hodhod.hotels.models.Hotel;
+import com.example.hodhod.hotels.network.ServiceGenerator;
+import com.example.hodhod.hotels.util.Resource;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModel;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+
+public class HotelViewModel extends ViewModel {
+
+    private static final String TAG = "HotelViewModel";
+
+    private MediatorLiveData<Resource<Hotel>> hotel;
+
+    public LiveData<Resource<Hotel>> getHotel() {
+
+        if (hotel == null) {
+            hotel = new MediatorLiveData<>();
+            hotel.setValue(Resource.loading((Hotel) null));
+
+            final LiveData<Resource<Hotel>> source = LiveDataReactiveStreams.fromPublisher(
+                    ServiceGenerator.getHotelApi()
+                            .getHotel()
+                            .onErrorReturn(new Function<Throwable, Hotel>() {
+                                @Override
+                                public Hotel apply(Throwable throwable) throws Exception {
```

Comment:
Nice idea but we don't need to map result , we can subscribe on Success/Fail directly.